### PR TITLE
Rover: Simplify processing

### PR DESCRIPTION
**Pull request by muramura**
I felt separation of treatment was inappropriate.
I integrated the processing.

### CODE REVIEW

File: APMrover2/commands_logic.cpp [hunk 1]
```diff
@@ -170,14 +170,16 @@ bool ModeAuto::verify_command(const AP_Mission::Mission_Command& cmd)
     case MAV_CMD_DO_SET_ROI:
     case MAV_CMD_DO_SET_REVERSE:
     case MAV_CMD_DO_FENCE_ENABLE:
-        return true;
```
Comment (amilcarlucas) [inline APMrover2/commands_logic.cpp]: I do not think this is an improvement

File: APMrover2/commands_logic.cpp [hunk 2]
```diff
@@ -170,14 +170,16 @@ bool ModeAuto::verify_command(const AP_Mission::Mission_Command& cmd)
...
+        break;
 
     default:
         // error message
         gcs().send_text(MAV_SEVERITY_WARNING, "Skipping invalid cmd #%i", cmd.id);
         // return true if we do not recognize the command so that we move on to the next command
-        return true;
```
Comment (amilcarlucas) [inline APMrover2/commands_logic.cpp]: I do not think this is an improvement

File: APMrover2/mode_guided.cpp [hunk 3]
```diff
@@ -31,11 +31,10 @@ void ModeGuided::update()
                 calc_steering_to_waypoint(_reached_destination ? rover.current_loc : _origin, _destination, _reversed);
                 calc_throttle(calc_reduced_speed_for_turn_or_distance(_reversed ? -_desired_speed : _desired_speed), true, true);
             } else {
-                if (rover.is_boat() && !start_loiter()) {
-                    stop_vehicle();
-                } else {
-                    stop_vehicle();
+                if (rover.is_boat()) {
```
Comment (muramura) [inline APMrover2/mode_guided.cpp]: https://github.com/ArduPilot/ardupilot/issues/10188
@rmackay9 san... I am changing the comment on this subject.

Comment (tajisoft) [reply APMrover2/mode_guided.cpp]: Thank you. I think there is no need to call stop_vehicle() even once..But it's just my opinion. Better way should be adapted.

**Comment by rmackay9** [comment]
I think i'm not going to merge this.  The top two changes are different from what we have but I don't think they really improve anything.  The change for Loiter isn't quite right I'm afraid because we shouldn't call stop_vehicle().

Thanks anyway!

**Pull request closed without merging**